inst/doc/pitfalls.Rmd

```diff
@@ -0,0 +1,51 @@
+
+* <sym>Common debugging methods don't work.</sym> If you build a `scipiper` target that runs function `A()` to create target `a`, setting RStudio checkpoints or `debug(A)` won't let you debug function `A` when you call `scmake('a')`. <sol>Temporarily insert a call to `browser()` within the function instead.</sol>
+
+* <sym>The GitHub repo is too big.</sym> You might be committing data files or large task remake files to the repo. <sol>In general, you can safely .gitignore all data files, the .remake/ folder, and (optionally) any task remake files.</sol>
+
+## Use of shared cache
+
+* <sym>Teammates aren't getting updates about available intermediate data files.</sym> When sharing intermediate data files with a team using a shared cache, the status information about those files is contained in (1) a \*.ind file promising the existence of the data file in the cache, and (2) a build/status/\*.yml file telling `remake` how to decide whether the \*.ind file is up to date. <sol>Always `git commit` the \*.ind and build/status/\*.yml files corresponding to all shared data files.</sol>
+
+* <sym>Teammates can't see updated objects.</sym> R objects can't be shared. <sol>To share objects, save them as .rds files and use the standard shared cache syntax.</sol>
+
+* <sym>Seemingly unnecessary builds of objects or files that a teammate already built.</sym> Those files may lack corresponding .ind files (and objects always lack .ind files), which are the key to avoiding rebuilds. With a shared cache approach, every data file or object not represented by an .ind file must be built locally by everybody who needs to build any target downstream of that file or object. <sol>For every file that takes a long time or special software to build: (1) use file rather than object targets, (2) define a corresponding .ind file and make downstream targets depend on the .ind file rather than the data file, (3) push the file to the shared cache, and (4) commit the .ind file to the repository.</sol>
+
+* <sym>There are more .ind files than I want to manage.</sym> Not every data file (or object) needs to be in the shared cache, and anything not in the shared cache is a candidate for also not having an .ind file. If (a) you're comfortable requiring everybody to build a file or object locally, or (b) there are no declared downstream dependencies of the file/object, then you can skip creating an .ind file for it. For example, you might have a task remake file whose execution (e.g., with `loop_tasks()`) depends on `X` and `Y` and produces a single output file `Z`. As long as the main remake file says that `loop_tasks()` creates `Z.ind` (not `Z`), you can have just one person run the entire task remake file, and your team can share the output, without using any .ind files within those tasks. <sol>Don't create .ind files for targets that don't need them.</sol>
```

Comment:
I would like to suggest adding a section about building targets while you're testing code changes and not changing the shared cache until you are ready.

Here's my suggestion:

`<sym>`I am working on code that changes data in the shared cache but I'm not ready to commit those changes, and teammates are being forced to rebuild.`</sym>` There are many times where you are locally working on code that changes data on the shared cache. You might want to test that you can build the target as you are developing without committing these new changes, and subsequent `.ind` and build/status/\*.yml files. However, when you change the shared cache by executing the code you are still working on locally, other teammates who are building targets that depend on that one will be forced to rebuild the target because their `.ind` and build files show that they are outdated.`<sol>`Use `options(scipiper.dry_put = TRUE)` to prevent files built from automatically being pushed to the shared cache. Remember to revert to `options(scipiper.dry_put = FALSE)` when you are ready to push your code changes.`</sol>`

Reply:
Oh, excellent idea! Thanks for the suggestion.